airflow/providers/amazon/aws/hooks/redshift_sql.py

```diff
@@ -104,7 +104,11 @@ def get_iam_token(self, conn: Connection) -> tuple[str, str, int]:
         port = conn.port or 5439
         # Pull the custer-identifier from the beginning of the Redshift URL
         # ex. my-cluster.ccdre4hpd39h.us-east-1.redshift.amazonaws.com returns my-cluster
-        cluster_identifier = conn.extra_dejson.get("cluster_identifier", conn.host.split(".")[0])
+        cluster_identifier = conn.extra_dejson.get("cluster_identifier", None)
+        if conn.host and not cluster_identifier:
+            cluster_identifier = conn.host.split(".")[0]
+        if not cluster_identifier and not conn.host:
```

Comment:
```suggestion
        if not cluster_identifier:
```
Not to eternize this or anything, but i think at this point we don't need to consider conn.host -- we've already processed it.   If at this point cluster_identifier is not populated, something is wrong, and we can raise with that message.

Reply:
removed it [here](https://github.com/apache/airflow/pull/31567/commits/6156ab5baf704f6f2117b3a8995476867e971417)